### PR TITLE
[cleanup]: refactor code to show the use of fakeClock explicitly in pkg/queue(#3380)

### PR DESCRIPTION
**Pull request by TusharMohapatra07**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Explicitly tells us where `fakeClock` is used in the code

#### Which issue(s) this PR fixes:
part of #3380 

#### Special notes for your reviewer:
This PR is work-in-progress. I've refactored the code inside `pkg/queue`. 

#### Release Note:
```release-note
NONE
```

### CODE REVIEW

**Comment by TusharMohapatra07** [comment]
@mbobrovskyi do we have to refactor `manager_test.go` inside `pkg/queue`. It has some instances of `time.Now()` here and there but those `now` variable have been passed into `Creation()` which takes in something of type `time.Time`

**Comment by mbobrovskyi** [comment]
Regarding this PR, I didn’t quite understand what changed here?

**Comment by TusharMohapatra07** [comment]
@mbobrovskyi nothing actually, the code was already using fakeClock, i just made it explicit inorder to keep the code uniform throughout wherever fakClock is used.

**Comment by mbobrovskyi** [comment]
I can see problems on:
- https://github.com/kubernetes-sigs/kueue/blob/main/pkg/workload/admissionchecks.go
- https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/admissionchecks/multikueue/workload.go
- https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/core/clusterqueue_controller.go
- https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobs/pod/pod_controller.go

**Comment by mimowo** [comment]
cc @mbobrovskyi @PBundyra

**Comment by mimowo** [comment]
@TusharMohapatra07 this PR is just a cleanup / formatting. I'm ok-ish with it as you make the code consistent with other places in the, like [here](https://github.com/kubernetes-sigs/kueue/blob/19bf982b09aa7f5c630d8b03e82408e4e8f20ca8/pkg/queue/cluster_queue_test.go#L298-L300). 

However, update description:
- say "Part of #3380"-don't use "Fixes", because then the robot will close the issue which is not completed by this PR. 
-  don't use "[fix]" in the title as this could be misleading
- instead of `/kind feature and /kind flake` say `/kind cleanup`

/remove-kind flake
/remove-kind feature
/kind cleanup

**Comment by TusharMohapatra07** [comment]
@mimowo done

**Comment by TusharMohapatra07** [comment]
Also do specify if i have to do anything with `manager_test.go`. I'll remove WIP if there's nothing to do there.

**Comment by mimowo** [comment]
I'm ok to merge it as is

**Comment by mbobrovskyi** [comment]
> Also do specify if i have to do anything with manager_test.go. I'll remove WIP if there's nothing to do there.

The `manager_test.go` looks good.

Please focus on https://github.com/kubernetes-sigs/kueue/pull/3478#issuecomment-2460596505 first. And we will see which tests should be fixed.

**Comment by mbobrovskyi** [comment]
/lgtm

Thanks!

**Comment by TusharMohapatra07** [comment]
@mbobrovskyi Yes, I'll be working on `blob/main` next.

File: pkg/queue/cluster_queue_test.go [hunk 1]
```diff
@@ -179,7 +180,9 @@ func Test_PushOrUpdate(t *testing.T) {
 
 func Test_Pop(t *testing.T) {
 	now := time.Now()
-	cq := newClusterQueueImpl(defaultOrdering, testingclock.NewFakeClock(now))
+	fakeClock := testingclock.NewFakeClock(now)
+
+	cq := newClusterQueueImpl(defaultOrdering, fakeClock)
```
Comment (tenzen-y) [inline pkg/queue/cluster_queue_test.go:185]: I was not sure the benefit such refactoring.
What is the reason that we need to additional malloc to new variable?

This seems just increase memory allocation.

**Comment by TusharMohapatra07** [comment]
@tenzen-y with reference to #3421, either we could pass in testingclock.NewFakeClock(now) directly in both the places. But i think that it's easier to build a mindmap by having fakeclock with time.Now() in the code. Idk, I'm just biased. I need the opinion of maintainers for this one.

**Comment by tenzen-y** [comment]
> @tenzen-y with reference to #3421, either we could pass in testingclock.NewFakeClock(now) directly in both the places. But i think that it's easier to build a mindmap by having fakeclock with time.Now() in the code. Idk, I'm just biased. I need the opinion of maintainers for this one.

We usually declare the variable only when the performed function is lack of context like

```
verified := hasValidAA() && hasValidBB
```

However, for this PR, `testingclock.NewFakeClock` has already enough context since the package name is clock of testing then new (initialize) fakeClock.

If we justify such a redeclaration, we will justify all of the variable declarations. So, I would propose reverting this PR.

**Comment by TusharMohapatra07** [comment]
@tenzen-y Sure. Also, Iam extremely sorry for the mess. I should've asked somebody before doing this.

**Comment by TusharMohapatra07** [comment]
> No worries :)
I've reverted the changes, should i create another PR for it ? Is there any way to make a commit to this PR ?

**Comment by mimowo** [comment]
@tenzen-y I'm ok to revert if you prefer to do so.

**Comment by tenzen-y** [comment]
> @tenzen-y Sure. Also, Iam extremely sorry for the mess. I should've asked somebody before doing this.

Yeah, you can just push this bottom.
Screenshot 2024-11-07 at 23 41 27

**Pull request merged**